src/Middleware/StaticFiles/src/StaticFileMiddleware.cs

```diff
@@ -95,52 +95,57 @@ public async Task Invoke(HttpContext context)
-                {
-                    case StaticFileContext.PreconditionState.Unspecified:
-                    case StaticFileContext.PreconditionState.ShouldProcess:
-                        if (fileContext.IsHeadMethod)
-                        {
-                            await fileContext.SendStatusAsync(Constants.Status200Ok);
-                            return;
-                        }
+                return ServeStaticFile(context, fileContext);
+            }
 
-                        try
-                        {
-                            if (fileContext.IsRangeRequest)
-                            {
-                                await fileContext.SendRangeAsync();
-                                return;
-                            }
+            return _next(context);
+        }
 
-                            await fileContext.SendAsync();
-                            _logger.FileServed(fileContext.SubPath, fileContext.PhysicalPath);
-                            return;
-                        }
-                        catch (FileNotFoundException)
+        private async Task ServeStaticFile(HttpContext context, StaticFileContext fileContext)
+        {
+            fileContext.ComprehendRequestHeaders();
+            switch (fileContext.GetPreconditionState())
+            {
+                case StaticFileContext.PreconditionState.Unspecified:
+                case StaticFileContext.PreconditionState.ShouldProcess:
+                    if (fileContext.IsHeadMethod)
+                    {
+                        await fileContext.SendStatusAsync(Constants.Status200Ok);
+                        return;
+                    }
+
+                    try
+                    {
+                        if (fileContext.IsRangeRequest)
                         {
-                            context.Response.Clear();
+                            await fileContext.SendRangeAsync();
+                            return;
                         }
-                        break;
-                    case StaticFileContext.PreconditionState.NotModified:
-                        _logger.FileNotModified(fileContext.SubPath);
-                        await fileContext.SendStatusAsync(Constants.Status304NotModified);
-                        return;
 
-                    case StaticFileContext.PreconditionState.PreconditionFailed:
-                        _logger.PreconditionFailed(fileContext.SubPath);
-                        await fileContext.SendStatusAsync(Constants.Status412PreconditionFailed);
+                        await fileContext.SendAsync();
+                        _logger.FileServed(fileContext.SubPath, fileContext.PhysicalPath);
                         return;
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        context.Response.Clear();
+                    }
+                    break;
```

Comment:
This case used to fall though to next but now returns OK. It's causing test failures.

Reply:
Ah, good spot, added